pkg/k8s/watchers/watcher.go

```diff
@@ -305,57 +373,17 @@ func (k *K8sWatcher) InitK8sSubsystem(ctx context.Context) <-chan struct{} {
 	}
 
 	go func() {
-		log.Info("Waiting until all pre-existing resources related to policy have been received")
-		// Wait only for certain caches, but not all!
-		// We don't wait for nodes synchronization.
-		k.WaitForCacheSync(
-			// To perform the service translation and have the BPF LB datapath
-			// with the right service -> backend (k8s endpoints) translation.
-			K8sAPIGroupServiceV1Core,
-			// To perform the service translation and have the BPF LB datapath
-			// with the right service -> backend (k8s endpoints) translation.
-			K8sAPIGroupEndpointV1Core,
-			K8sAPIGroupEndpointSliceV1Beta1Discovery,
-			K8sAPIGroupEndpointSliceV1Discovery,
-			// We need all network policies in place before restoring to make sure
-			// we are enforcing the correct policies for each endpoint before
-			// restarting.
-			k8sAPIGroupCiliumNetworkPolicyV2,
-
-			k8sAPIGroupCiliumClusterwideNetworkPolicyV2,
-			// We we need to know about all other nodes
-			k8sAPIGroupCiliumNodeV2,
-			// We need all network policies in place before restoring to make sure
-			// we are enforcing the correct policies for each endpoint before
-			// restarting.
-			k8sAPIGroupNetworkingV1Core,
-			// Namespaces can contain labels which are essential for endpoints
-			// being restored to have the right identity.
-			k8sAPIGroupNamespaceV1Core,
-			// Pods can contain labels which are essential for endpoints
-			// being restored to have the right identity.
-			K8sAPIGroupPodV1Core,
-			// We need to know about active local redirect policy services
-			// before BPF LB datapath is synced.
-			k8sAPIGroupCiliumLocalRedirectPolicyV2,
-			// We need to know the node labels to populate the host endpoint
-			// labels.
-			k8sAPIGroupNodeV1Core,
-		)
-		// CiliumEndpoint is used to synchronize the ipcache, wait for
-		// it unless it is disabled
-		if !option.Config.DisableCiliumEndpointCRD {
-			k.WaitForCacheSync(k8sAPIGroupCiliumEndpointV2)
-		}
+		log.Info("Waiting until all pre-existing resources have been received")
+		k.WaitForCacheSync(resources...)
 		close(cachesSynced)
 	}()
 
 	go func() {
 		select {
 		case <-cachesSynced:
-			log.Info("All pre-existing resources related to policy have been received; continuing")
```

Comment:
These log messages mentioning this is "related to policy" have been wrong for some time now since we wait for just about every type of resource, so I just dropped those bits from the messages.